junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/JupiterTestDescriptor.java

```diff
@@ -81,7 +81,7 @@
 
 	@Override
 	public boolean isLeaf() {
-		return !isContainer();
+		return isTest();
```

Comment:
I doubt this assumption is really true for all tree leaves. Their might be leaves that are by definition containers, e.g. dynamic tests.